### ISSUE TITLE
Errors of conversion from ONNX (opset = 9) model to NCNN-model (param / bin)

### ISSUE THREAD

**Issue by AlexeyAB** [Tencent/ncnn#1942]
@nihui Hi,

I have an ONNX-model (opset = 9) with NCHW layout.

I try to convert ONNX-model to NCNN-model (param / bin), but I get these errors: 
`tools/onnx/onnx2ncnn model.onnx model.param model.bin`

Output:
```
Shape not supported yet!
Gather not supported yet!
  # axis=0
Cast not supported yet!
  # to=1
Cast not supported yet!
  # to=1
Shape not supported yet!
Gather not supported yet!
  # axis=0
Cast not supported yet!
  # to=1
Cast not supported yet!
  # to=1
Unsupported unsqueeze axes !
Unsupported unsqueeze axes !
Cast not supported yet!
  # to=1
Shape not supported yet!
Cast not supported yet!
  # to=1
Unknown data type 0
Unsupported Upsample scales !
...
```

It generated model.param / bin, fut shows an error when I try to run NCNN
`layer Shape not exists or registered`

----

If I first run `onnx-simplifier` and then run converter, then the conversion succeeds.
But when I run this model.param / bin by using NCNN, the output of inference doesn't match to output in Pytorch / TensorFlow very much.

----

* Is there any way to fix this error?
* Does NCNN supports NCWH-layouts?
* Should I use ONNX-opset=9 or another opset version?
* Should I always use onnx-simplifier?

**Comment by nihui** [comment]
* onnx-simplifier is required usually
* ncnn supports NCHW-layout
* onnx opset version 9 is fine

https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx

If the output differs from what the original framework produces, check this
https://github.com/Tencent/ncnn/wiki/FAQ-ncnn-produce-wrong-result

**Comment by AlexeyAB** [comment]
@nihui Thanks, the issue is solved!

1. I should use `onnx-simplifier`

2. I just forgot to divide `/255`, since mean/var normalization is built in my onnx-model:
```cpp
    float mean[3] = { 0, 0, 0 };
    float norm[3] = { 1/255.f, 1/255.f, 1/255.f };
    in.substract_mean_normalize(mean, norm);
```

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
